### PR TITLE
implement change password endpoint #407

### PR DESCRIPTION
**Pull request by MohitKS5**
Signed-off-by: MohitKS5 <mohitkumarsingh907@gmail.com>
in reponse to #407 
### Bullet Points
* Implements `/password` endpoint
* UIAA auth check to be included from #413 (if merged) or some other refactor.

##### Testing
* pass username in addition to spec sheet params till whoami functions are available. I dont see a point in copy pasting the code of an entire endpoint.

### CODE REVIEW

File: src/github.com/matrix-org/dendrite/clientapi/routing/password.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// Copyright 2017 Vector Creations Ltd
+// Copyright 2017 New Vector Ltd
+//
```
Comment (krombel) [inline src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: This file (and its contents) got created in 2018. And now we just have New Vector

File: src/github.com/matrix-org/dendrite/clientapi/routing/password.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	logger := util.GetLogger(req.Context())
+	logger.WithFields(log.Fields{
+		"username":   r.Username,
+		"auth.type":  r.Auth.Type,
+		"session_id": r.Auth.Session,
+	}).Info("Processing password request")
+
+	// Todo go through User Interactive Auth uncomment references in #413  and
+	// authflows defined in cfg.derieved like that for registration
+
```
Comment (krombel) [inline src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: Typo "derieved"

File: src/github.com/matrix-org/dendrite/clientapi/routing/password.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	//
+	//	return completeUpdation(
+	//		req.Context(),
+	//		accountDB,
+	//		r.Username,
+	//		r.Password,
+	//		res.AppserviceID,
+	//		r.InitialDisplayName)
+	//}
+	//return jsonRes
```
Comment (krombel) [inline src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: I would say to not add code as a comment. Especially when you are done with implementing.

Comment (MohitKS5) [reply src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: Actually that code was to be added when #413 is merged or at least its motive is served. I wrote it as a draft for reviewers. Will remove it if u say.

File: src/github.com/matrix-org/dendrite/clientapi/routing/password.go (hunk 2 above)
Comment (MohitKS5) [reply src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: I apologize for being negligent in comments :\. Will change to `cfg.Derived`

File: src/github.com/matrix-org/dendrite/clientapi/routing/password.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+// ChangePassword implements a /password request.
+func ChangePassword(
+	req *http.Request,
+	accountDB *accounts.Database,
+	cfg *config.Dendrite,
+) util.JSONResponse {
+
+	// Todo username and password aren't in request as per spec. Use whoami referenced in #411
```
Comment (anoadragon453) [inline src/github.com/matrix-org/dendrite/clientapi/routing/password.go]: Todo's are typically in the fashion `TODO: ...` btw, and some IDEs require explicitly that.

**Comment by Kegsay** [comment]
This PR was created in 2018 and has now been marked as `stale`. Please can someone resurrect this or else it will be closed in 1 week.

**Pull request closed without merging**